scheduler/src/main/java/gov/nasa/jpl/aerie/scheduler/Problem.java

```diff
@@ -12,37 +16,78 @@
    *
    * @param mission IN the mission model that this problem is based on
    */
-  public Problem(MissionModelWrapper mission) {
-    if (mission == null) {
-      throw new IllegalArgumentException(
-          "creating problem descriptor with null mission model");
+  public Problem(MissionModel<?> mission, PlanningHorizon planningHorizon) {
+    this.missionModel = mission;
+    this.initialPlan = new PlanInMemory();
+    this.planningHorizon = planningHorizon;
+    //TODO: find cleaner way to handle built-in act types
+    //TODO: find cleaner way to handle windows in general
+    //special activity for displaying valid windows in visualization
+    add(new ActivityType("Window"));
+
+    //include special activity type for marking plan horizon
+    add(new ActivityType("HorizonMarker"));
+
+    //add all activity types known to aerie to scheduler index
+    //TODO: reduce duplicate activity type abstractions between aerie and scheduler
+    if( missionModel != null ) {
+      for(var taskType : missionModel.getTaskSpecificationTypes().entrySet()){
+        this.add(new ActivityType(taskType.getKey(), taskType.getValue()));
+      }
     }
-    this.mission = mission;
-    this.initialPlan = new PlanInMemory(mission);
+
+    simulationFacade = new SimulationFacade(planningHorizon, missionModel);
   }
 
   /**
    * the mission model that this problem is based on
    */
-  protected MissionModelWrapper mission;
 
+  private final MissionModel<?> missionModel;
+
+  private final SimulationFacade simulationFacade;
+
+  //should be accessible inside problem definition
+  protected final PlanningHorizon planningHorizon;
```

Comment:
Might want to move the variable declaration to the beginning of the class

Reply:
I have moved them